js/binancetr.js

```diff
@@ -0,0 +1,308 @@
+        //                            {"filterType":"PERCENT_PRICE","multiplierUp":5,"multiplierDown":0.2,"avgPriceMins":"5","applyToMarket":false},
+        //                            {"filterType":"LOT_SIZE","minQty":"0.10000000","maxQty":"9222449.00000000","stepSize":"0.10000000","applyToMarket":false},
+        //                            {"filterType":"MIN_NOTIONAL","avgPriceMins":"5","minNotional":"10.00000000","applyToMarket":true},
+        //                            {"filterType":"ICEBERG_PARTS","applyToMarket":false,"limit":"10"},
+        //                            {"filterType":"MARKET_LOT_SIZE","minQty":"0.00000000","maxQty":"191987.27930555","stepSize":"0.00000000","applyToMarket":false},
+        //                            {"filterType":"MAX_NUM_ORDERS","applyToMarket":false},
+        //                            {"filterType":"MAX_NUM_ALGO_ORDERS","applyToMarket":false,"maxNumAlgoOrders":"5"}
+        //                        ],
+        //                        "orderTypes":[
+        //                            "LIMIT",
+        //                            "LIMIT_MAKER",
+        //                            "MARKET",
+        //                            "STOP_LOSS_LIMIT",
+        //                            "TAKE_PROFIT_LIMIT"
+        //                            ],
+        //                        "icebergEnable":1,
+        //                        "ocoEnable":1,
+        //                        "spotTradingEnable":1,
+        //                        "marginTradingEnable":0,
+        //                        "permissions":["SPOT"]
+        //                    },
+        //                ]
+        //            }
+        //        }
+        //
+        const result = [];
+        const data = this.safeValue (response, 'data', {});
+        const list = this.safeValue (data, 'list', []);
+        for (let i = 0; i < list.length; i++) {
+            const market = list[i];
+            // const type = this.safeInteger(market, 'type'); // Important to see endpoint for this market
+            const id = this.safeString (market, 'symbol');
+            const filters = this.safeValue (market, 'filters');
+            const filter0 = this.safeValue (filters, 0);
+            const filter2 = this.safeValue (filters, 2);
+            const permissions = this.safeValue (market, 'permissions');
+            const minPrice = this.safeValue (filter0, 'minPrice');
+            const maxPrice = this.safeValue (filter0, 'maxPrice');
+            const minQty2 = this.safeValue (filter2, 'minQty');
+            const maxQty2 = this.safeValue (filter2, 'maxQty');
+            const baseId = this.safeString (market, 'baseAsset');
+            const quoteId = this.safeString (market, 'quoteAsset');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const spotPer = permissions.indexOf ('SPOT') > -1;
```

Comment:
`indexOf()` won't work on arrays in Py/PHP, we have to loop over `permissions` here.
Also, please, avoid abbreviations like `spotPer` – we don't have to save bytes on variable names.

Reply:
I only used indeOf because of a suggestion of frosty just above. Anyway, no problem, I changed it to a loop as you suggest.
Also changed this and other variable names to a more descriptive longer name.
Done.

Reply:
> I only used indeOf because of a suggestion of frosty just above. Anyway, no problem, I changed it to a loop as you suggest.

Yep, indexOf is for strings only.

> Also changed this and other variable names to a more descriptive longer name.

Cool, thank you!

> Done.

Can you, please, update this PR (push the latest edits)? We need to make sure that is passes the tests:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-build--run-tests-on-your-local-machine

Reply:
Ok, I'll push itnow, but it have to change the rest of your code reviews. 
I confirm it builds ok on my local machine.

Reply:
@criadoperez great, thx!

Reply:
latest changes have broken the build again. I'm fixing them...